scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroTap.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.parquet.avro
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.io.Tap
+import com.spotify.scio.parquet.{BeamInputFile, ParquetConfiguration}
+import com.spotify.scio.util.ScioUtil
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.io._
+import org.apache.parquet.avro.AvroParquetReader
+
+import scala.jdk.CollectionConverters._
+import scala.reflect.ClassTag
+
+final case class ParquetAvroTap[A, T: ClassTag: Coder](
+  path: String,
+  params: ParquetAvroIO.ReadParam[A, T]
+) extends Tap[T] {
+  override def value: Iterator[T] = {
+    val filePattern = ScioUtil.filePattern(path, params.suffix)
+    val conf = ParquetConfiguration
+      .ofNullable(params.conf)
+    params.setReadSchemas(params)
```

Comment:
do we want to do this ?

Reply:
probably was tired when committed this, I am changing the approach